### ISSUE TITLE
Pseudo-element selector should not be accepted when there are spaces between colon and pseudo-element name

### ISSUE THREAD

**Issue by upsuper** [servo/servo#15335]
Servo fails with the following test:
```html
<!DOCTYPE html>
<style>
div::before {
  color: green;
  content: "PASS";
}
div:: before {
  color: red;
  content: "FAIL";
}
</style>
<div></div>
```

It is probably an issue in the cssparser. And this could probably be an easy bug.

cc @SimonSapin

**Comment by SimonSapin** [comment]
In https://github.com/servo/rust-selectors, in `src/parser.rs`, in the `test_parsing` function, a new test should be added near the existing `parse("::before")` test to check that `parse(":: before")` returns `Err(())` as it should. Add the URL of this issue in a comment.

Once that test has been observed failing, the bug should be fixed. It’s in the `parse_one_simple_selector` function of the same file. In the case where two consecutive `Token::Colon` are found, the code currently calls `input.next()` (and then `parse_pseudo_element` if it finds an identifier). That calls need to be changed to `input.next_including_whitespace()`. That change should make the new test pass.

**Comment by atheed** [comment]
I'd like to take this.

**Comment by SimonSapin** [comment]
Go ahead!

**Comment by SimonSapin** [comment]
https://github.com/servo/rust-selectors/pull/103 is merged and v0.15.2 on crates.io. When updating Servo we should probably include the test case above.

**Comment by upsuper** [comment]
It seems this has been fixed. Closing.
